File: runtime/flutter/services/restorationBucket.ts
export interface IRestorationBucket {
    getDebugOwner: () => Object | undefined;
    getIsReplacing: () => boolean;
    getRestorationId: () => string;
    read: <P>(restorationId: string) => P | undefined;
    write: <P>(restorationId: string, value: P) => void;
    remove: <P>(restorationId: string) => P | undefined;
    contains: (restorationId: string) => boolean;
    claimChild: (
        restorationId: string,
        props?: { debugOwner?: Object | undefined }
    ) => IRestorationBucket;
    adoptChild: (child: IRestorationBucket) => void;
    rename: (newRestorationId: string) => void;
    dispose: () => void;
    toString: () => string;
}
export class RestorationBucket {
    public static _childrenMapKey = "c";
    public static _valuesMapKey = "v";
    private readonly _dart_getDebugOwner: () => Object | undefined =
        undefined as any;
    private readonly _dart_getIsReplacing: () => boolean = undefined as any;
    private readonly _dart_getRestorationId: () => string = undefined as any;
    private readonly _dart_read: <P>(restorationId: string) => P | undefined =
        undefined as any;
    private readonly _dart_write: <P>(restorationId: string, value: P) => void =
        undefined as any;
    private readonly _dart_remove: <P>(restorationId: string) => P | undefined =
        undefined as any;
    private readonly _dart_contains: (restorationId: string) => boolean =
        undefined as any;
    private readonly _dart_claimChild: (
        restorationId: string,
        props?: { debugOwner?: Object | undefined }
    ) => IRestorationBucket = undefined as any;
    private readonly _dart_adoptChild: (child: IRestorationBucket) => void =
        undefined as any;
    private readonly _dart_rename: (newRestorationId: string) => void =
        undefined as any;
    private readonly _dart_dispose: () => void = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    public getDebugOwner(): Object | undefined {
        return this._dart_getDebugOwner();
    }
    public getIsReplacing(): boolean {
        return this._dart_getIsReplacing();
    }
    public getRestorationId(): string {
        return this._dart_getRestorationId();
    }
    public read<P>(restorationId: string): P | undefined {
        return this._dart_read(restorationId);
    }
    public write<P>(restorationId: string, value: P): void {
        return this._dart_write(restorationId, value);
    }
    public remove<P>(restorationId: string): P | undefined {
        return this._dart_remove(restorationId);
    }
    public contains(restorationId: string): boolean {
        return this._dart_contains(restorationId);
    }
    public claimChild(
        restorationId: string,
        props?: { debugOwner?: Object | undefined }
    ): IRestorationBucket {
        return this._dart_claimChild(restorationId, props);
    }
    public adoptChild(child: IRestorationBucket): void {
        return this._dart_adoptChild(child);
    }
    public rename(newRestorationId: string): void {
        return this._dart_rename(newRestorationId);
    }
    public dispose(): void {
        return this._dart_dispose();
    }
    public toString(): string {
        return this._dart_toString();
    }
}

File: lib/cfr/builtins/libs/flutter/services/restorationBucket.dart
import 'dart:core' as _fac9;

import 'package:flutter/src/services/restoration.dart' as _6d4f;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedRestorationBucket
    extends _36c2.VMManagedBox<_6d4f.RestorationBucket> {
  VMManagedRestorationBucket(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['getDebugOwner'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      final returnValue = vmObject.debugOwner;
      if (returnValue != null) {
        return [
          _36c2.maybeBoxObject<_fac9.Object?>(
              object: returnValue,
              hydroState: hydroState,
              table: _36c2.HydroTable()),
        ];
      }
      return [];
    });
    table['getIsReplacing'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.isReplacing,
      ];
    });
    table['getRestorationId'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.restorationId,
      ];
    });
    table['read'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      final returnValue = vmObject.read(luaCallerArguments[1]);
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['write'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      vmObject.write(luaCallerArguments[1], luaCallerArguments[2]);
      return [];
    });
    table['remove'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      final returnValue = vmObject.remove(luaCallerArguments[1]);
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['contains'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.contains(luaCallerArguments[1]),
      ];
    });
    table['claimChild'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_6d4f.RestorationBucket>(
            object: vmObject.claimChild(luaCallerArguments[1],
                debugOwner: _36c2
                    .maybeUnBoxAndBuildArgument<_fac9.Object?, _fac9.dynamic>(
                        luaCallerArguments.length >= 3
                            ? luaCallerArguments[2]['debugOwner']
                            : null,
                        parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['adoptChild'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      vmObject.adoptChild(_36c2.maybeUnBoxAndBuildArgument<
          _6d4f.RestorationBucket,
          _fac9.dynamic>(luaCallerArguments[1], parentState: hydroState));
      return [];
    });
    table['rename'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      vmObject.rename(luaCallerArguments[1]);
      return [];
    });
    table['dispose'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      vmObject.dispose();
      return [];
    });
    table['toString'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.toString(),
      ];
    });
  }

  final _36c2.HydroTable table;

  final _36c2.HydroState hydroState;

  final _6d4f.RestorationBucket vmObject;
}

void loadRestorationBucket(
    {required _36c2.HydroState hydroState, required _36c2.HydroTable table}) {
  _36c2.registerBoxer<_6d4f.RestorationBucket>(boxer: (
      {required _6d4f.RestorationBucket vmObject,
      required _36c2.HydroState hydroState,
      required _36c2.HydroTable table}) {
    return VMManagedRestorationBucket(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}
